pkg/redis/RedisConnectionFactory.php

```diff
@@ -50,13 +50,18 @@ public function __construct($config = 'redis:')
         }
 
         $this->config = array_replace($this->defaultConfig(), $config);
+        if (isset($this->config['vendor'])) {
```

Comment:
Could you explain what this `if ... else` for? The `defaultConfig` method returns `vendor` option, so it has to be present all the time.

Reply:
good point! I added that line because of the check in parseDsn() method (there the parameter is not present for some of the tests as it is loaded only from the dsn string). 
in __construct the check is indeed not required. I'll remove this line.